### PR TITLE
SYRCoSE article

### PR DESCRIPTION
**Pull request by volodeyka**

### CODE REVIEW

**Comment by eupp** [comment]
Do we need the following files?

1. `SYRCoSE/IEEEtran_HOWTO.pdf`
2. `SYRCoSE/conference_101719.pdf`
3. ` SYRCoSE/conference_101719.tex`
4. `SYRCoSE/fig1.png`

Also, it would be nice to create a top-level dir `papers` and put `SYRCoSE` folder there.

**Comment by volodeyka** [comment]
> Do we need the following files?

@dmitromikh?

**Comment by dmitromikh** [comment]
> > Do we need the following files?
> 
> @dmitromikh?

Ah, they are from the Overleaf template. So, we really do not need these files.

File: papers/SYRCoSE/related.tex [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+to mechanize the classical theory of event structures 
+in proof assistants. 
+ 
+More recently event structures were applied 
+in the context of relaxed memory models~%
+\cite{Jeffrey-Riely:LICS16, PichonPharabod-Sewell:POPL16, Chakraborty-Vafeiadis:POPL19, Moiseenko-al:ECOOP20}.
+Among this line of work, we are aware of only one paper~\cite{Moiseenko-al:ECOOP20}
+that was accompanied with the mechanization in a proof assistant. 
+The authors formalized \weakestmo~\cite{Chakraborty-Vafeiadis:POPL19} 
+memory memory model in \coq. However, this memory model uses
```
Comment (dmitromikh) [inline papers/SYRCoSE/related.tex:17]:
```suggestion
memory model in \coq. However, this memory model uses 
```

File: papers/SYRCoSE/regmachine-eventstruct.tex [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+  Then $\lessdot$ is covering relation \wrt $\leqslant$ whenever 
+  $x \lessdot y$ is true if and only if $x < y$ and 
+  there is no $z$ \sth $x < z$ and $z < y$.
+  A (non-deterministic) function $f : A \fun \pwset{A}$ is a covering function if
+  its corresponding relation, \ie ${\frel{f} \defeq \set{\tup{x, y} ~|~ y \in f(x)}}$, 
+  is a covering relation.
+\end{definition}
+ 
+We use the inverse covering function because it is 
+more convinient in our setting. Indeed, the semanics adds new event
```
Comment (dmitromikh) [inline papers/SYRCoSE/regmachine-eventstruct.tex:61]:
```suggestion
more convenient in our setting. Indeed, the semantics adds a new event
```

File: papers/SYRCoSE/regmachine-eventstruct.tex [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+at each step. Then it is convinient to require that in addition
```
Comment (dmitromikh) [inline papers/SYRCoSE/regmachine-eventstruct.tex:62]:
```suggestion
at each step. Then it is convenient to require that in addition 
```

File: papers/SYRCoSE/regmachine-eventstruct.tex [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+label, $\lPO$ and $\lRF$ predecessors. 
+The \ESIdleRule handles the case when a thread of 
+the register machine performs an internal step 
+without any side effect. 
+It chooses an event $e$ together with 
+the thread state $s$ corresponding to it
+and performs one step reduction to a new state $s'$.
+It then updates the mapping of events to thread states.   
+The last two rules \ESStoreRule and \ESLoadRule
+correspond to store and load perfromed by some thread.
```
Comment (dmitromikh) [inline papers/SYRCoSE/regmachine-eventstruct.tex:134]:
```suggestion
correspond to store and load performed by some thread.  
```

**Pull request merged**